app/src/main/java/de/simon/dankelmann/bluetoothlespam/Handlers/AdvertisementSetQueueHandler.kt

```diff
@@ -195,8 +214,15 @@ class AdvertisementSetQueueHandler(
         }
 
         if (_active) {
-            val preparedSet = prepareAdvertisementSet(nextSet)
-            _advertisementService.startAdvertisement(preparedSet)
+            // Only advertise if the set is checked
+            if (nextSet.isChecked) {
+                val preparedSet = prepareAdvertisementSet(nextSet)
+                _advertisementService.startAdvertisement(preparedSet)
+            } else {
+                // If the set is not checked, immediately move to the next one
```

Comment:
Imho it should not be the job of `fun advertiseNextAdvertisementSet` to handle skipping items. Just by the names, it makes more sense for `fun selectNextAdvertisementSet` to contain that skipping logic.